applications/ls/src/lib.rs

```diff
@@ -49,7 +58,12 @@ pub fn main(args: Vec<String>) -> isize {
     // Navigate to the path specified by first argument
     match path.get(&curr_wd) {
         Some(FileOrDir::Dir(dir)) => {
-            print_children(&dir);
+            if size_option {
+                print_children(&dir, true);
```

Comment:
Same comment here, no need to duplicate code, just pass in `size_option` directly as an argument.